lib/SyTest/Federation/Datastore.pm

```diff
@@ -246,4 +251,71 @@ sub next_room_id
    return sprintf "!%d:%s", $self->{next_room_id}++, $self->server_name;
 }
 
+=head2 get_room
+
+   $room = $store->get_room( $room_id )
+
+Returns a L<SyTest::Federation::Room> having the given ID, if one exists, or
+C<undef> if not.
+
+=cut
+
+sub get_room
+{
+   my $self = shift;
+   my ( $room_id ) = @_;
+
+   return $self->{rooms_by_id}{$room_id};
+}
+
+=head2 create_room
+
+   $room = $store->create_room( creator => $creator, [ alias => $alias ] )
+
+Creates a new L<SyTest::Federation::Room> instance with a new room ID and
+stores it in the data store. It creates the initial room events using the
+given C<creator> user ID. It associates the optional C<alias> if supplied.
+
+=cut
+
+sub create_room
+{
+   my $self = shift;
+   my %args = @_;
+
+   my $creator = $args{creator};
+
+   my $room = SyTest::Federation::Room->new(
+      datastore => $self,
+   );
+
+   $room->create_initial_events(
+      creator => $creator,
+   );
+
+   $self->{rooms_by_id}{ $room->room_id } = $room;
+
+   $self->{room_aliases}{ $args{alias} } = $room->room_id
+      if $args{alias};
+
+   return $room;
+}
+
+=head2 get_alias
+
+   $room_id = $store->get_alias( $room_alias )
+
+Returns the room ID associated with the given room alias, if one exists, or
+C<undef> if not.
+
+=cut
+
+sub get_alias
```

Comment:
Maybe lookup_alias? Or get_room_by_alias? I generally expect get functions to return the thing named after get, but this doesn't get an alias, it gets a room.